src/views/components/UserProfile/index.tsx

```diff
@@ -0,0 +1,221 @@
+import React from "react";
+import {
+  Button,
+  Container,
+  Form,
+  Input,
+  Message,
+  Select
+} from "semantic-ui-react";
+import options from "./helper";
+
+export interface Opt {
+  key: string;
+  text: string;
+  value: string;
+}
+
+export interface LoggedIn {
+  userName: string;
+  role: string;
+}
+
+export interface Profile {
+  userName: string;
+  password: string;
+  email: string;
+  role: string;
+}
+
+export interface Props {
+  roleOptions: Opt[];
+  editUser: Profile;
+  loggedInUser: LoggedIn;
+  autoFocus: boolean;
+  onSubmit: (data: any) => Promise<void>;
+  search: boolean;
+  selection: boolean;
+  options: Opt[];
+  serviceOptions: Opt[];
+  branches: Opt[];
+  onChange: (e) => void;
+  deleteUserHandler: (e) => void;
+}
+
+export interface State {
+  editUser: Profile;
+  emailError?: string;
+}
+
+class UserProfile extends React.Component<Props, State> {
+  public state: State;
+
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      editUser: this.props.editUser,
+      emailError: null
+    };
+  }
+
+  public handleInput = event => {
+    const name = event.target.name;
+    const value = event.target.value;
+
+    const editUser = { ...this.state.editUser, [name]: value };
+    this.setState({ editUser });
+  };
+  // TODO: Method to be used in dropdown though do not how to get the event.target.value
+  // public handleDropdownInput = (event, { value }) => {
```

Comment:
the data object `onChange(event: SyntheticEvent, data: object)` has `name` & `value`. You should have attempted logging it.

```suggestion
   public handleDropdownInput = (_event, {name, value}) => {
    const editUser = { ...this.state.editUser, [name]: value}
    this.setState({ editUser });
  }
```